tests/execute/result/check_results/test.fmf

```diff
@@ -0,0 +1,84 @@
+summary: Tests for check results behaviour
+description: Verify that check results are correctly interpreted and affect test results
+
+/check-pass:
+    summary: Test with passing checks
+    test: echo "Test passed"
+    framework: shell
+    duration: 1m
+    check:
+        - how: dmesg
+          result: respect
+    # Expected outcome: PASS (test passes, check passes)
+
+/check-fail-respect:
+    summary: Test with failing dmesg check (respect)
+    test: echo "Fail Test Check Pattern" | sudo tee /dev/kmsg
```

Comment:
If the test needs `root` for their execution I would suggest to mark `/tests/execute/result/check` with `provision-only` and `provision-local` tags (optionally with `provision-virtual` as well) to make sure it gets the permissions and is not executed as part of the regular plans.

Reply:
Thanks. I was gonna ask about the root thing - Isn't there a better way to trigger a check failure that doesn't require root?

Reply:
I'm not aware of any. Plus, in order to actually execute the check (either avc or dmesg) root is still necessary, so I guess we do not have any other choice. And if we're under root we can drop the `sudo`, right?